### ISSUE TITLE
setup help on minion

### ISSUE THREAD

**Issue by brentp** [LooseLab/readfish#30]
hi, I'm trying to get this set up on a minion. I have this as far as getting `ru_unblock_all` to work on the playback. I am not able to get the example with selective unblock going. It looks like reads are never getting sent to minimap2. The stderrr looks like:
```
020-03-09 16:50:08,856 Manager /home/quinlan/.local/bin/ru_generators --device MN18894 --experiment-name RU Test basecall and map --toml /data/human_chr_selection.toml --log-file human_chr_selection.log
2020-03-09 16:50:08,856 Manager batch_size=512
2020-03-09 16:50:08,856 Manager cache_size=512
2020-03-09 16:50:08,856 Manager channels=[1, 512]
2020-03-09 16:50:08,856 Manager chunk_log=chunk_log.log
2020-03-09 16:50:08,856 Manager device=MN18894
2020-03-09 16:50:08,856 Manager dry_run=False
2020-03-09 16:50:08,856 Manager experiment_name=RU Test basecall and map
2020-03-09 16:50:08,856 Manager host=127.0.0.1
2020-03-09 16:50:08,856 Manager log_file=human_chr_selection.log
2020-03-09 16:50:08,856 Manager log_format=%(asctime)s %(name)s %(message)s
2020-03-09 16:50:08,856 Manager log_level=info
2020-03-09 16:50:08,856 Manager paf_log=paflog.log
2020-03-09 16:50:08,856 Manager port=9501
2020-03-09 16:50:08,856 Manager read_cache=AccumulatingCache
2020-03-09 16:50:08,856 Manager run_time=172800
2020-03-09 16:50:08,856 Manager throttle=0.1
2020-03-09 16:50:08,856 Manager toml=/data/human_chr_selection.toml
2020-03-09 16:50:08,857 Manager unblock_duration=0.1
2020-03-09 16:50:08,857 Manager workers=1
2020-03-09 16:50:08,860 Manager Initialising minimap2 mapper
2020-03-09 16:50:16,204 Manager Mapper initialised
2020-03-09 16:50:16,204 read_until_api_v2.main Client type: many chunk
2020-03-09 16:50:16,204 read_until_api_v2.main Cache type: AccumulatingCache
2020-03-09 16:50:16,204 read_until_api_v2.main Filter for classes: adapter and strand
2020-03-09 16:50:16,204 read_until_api_v2.main Creating rpc connection for device MN18894.
2020-03-09 16:50:16,480 read_until_api_v2.main Loaded RPC
2020-03-09 16:50:16,481 read_until_api_v2.main Signal data-type: int16
2020-03-09 16:50:16,482 Manager This experiment has 1 region on the flowcell
2020-03-09 16:50:16,482 Manager Using reference: /data/human_g1k_v38_decoy_phix.fasta.mmi
2020-03-09 16:50:16,483 Manager Region 'select_chr_21_22' (control=False) has 2 targets of which 2 are in the reference. Reads will be unblocked when classed as single_off or multi_off; sequenced when classed as single_on or multi_on; and polled for more data when classed as no_map or no_seq.
2020-03-09 16:50:16,484 Manager Creating 1 workers
2020-03-09 16:50:16,484 read_until_api_v2.main Processing started
2020-03-09 16:50:16,485 read_until_api_v2.main Sending init command, channels:1-512, min_chunk:0
```
and then it just stalls there.

Other potentially useful information:
+ I have started the run in minKNOW (both with and without minKNOW base-calling -- and trying an external guppy base-calling server)
+ if I enter a different PORT, then it fails as it can't connect.
+ here is the command I am using:
```
ru_generators --device $DEVICE \
              --port $PORT \
              --experiment-name "RU Test basecall and map" \
              --toml $TOML \
              --log-file $(basename $TOML .toml).log
```
+ In minKNOW I can see the run proceeding un-interrrupted (whereas with `ru_unblock_all` I can see the size distribution drop).

Anything else I can report or try to help diagnose?

**Comment by alexomics** [comment]
Could you paste your TOML file here? You can remove IP addresses/paths/ports

**Comment by alexomics** [comment]
Also, when terminating Read Until what’s the error/trace back?

**Comment by brentp** [comment]
Here is the toml:
```
[caller_settings]
config_name = "dna_r9.4.1_450bps_hac"
host = "127.0.0.1"
port = 5555

[conditions]
reference = "/data/human_g1k_v38_decoy_phix.fasta.mmi"

[conditions.0]
name = "select_chr_21_22"
control = false
min_chunks = 0
max_chunks = inf
targets = ["chr21", "chr22"]
single_on = "stop_receiving"
multi_on = "stop_receiving"
single_off = "unblock"
multi_off = "unblock"
no_seq = "proceed"
no_map = "proceed"
```
and here is the traceback when pressing ctrl+c on the process that was running `ru_generators`:
```
2020-03-10 15:33:15,340 read_until_api_v2.main Sending init command, channels:1-512, min_chunk:0


^C2020-03-11 08:50:15,198 Manager Caught ctrl-c, terminating workflow.
2020-03-11 08:50:15,203 read_until_api_v2.main Reset request received, shutting down...
2020-03-11 08:50:15,210 read_until_api_v2.main Reset signal received by action handler.
2020-03-11 08:50:15,252 read_until_api_v2.main Stopping processing of reads due to reset.
SIGINT received. Terminating server.
2020-03-11 08:50:15,397 read_until_api_v2.main Stream handler exited successfully.
2020-03-11 08:50:15,405 Manager EXCEPT
Traceback (most recent call last):
  File "/home/quinlan/.local/lib/python3.6/site-packages/ru/ru_gen.py", line 408, in run_workflow
    res = result.get(3)
  File "/usr/lib/python3.6/multiprocessing/pool.py", line 644, in get
    raise self._value
  File "/usr/lib/python3.6/multiprocessing/pool.py", line 119, in worker
    result = (True, func(*args, **kwds))
  File "/home/quinlan/.local/lib/python3.6/site-packages/ru/ru_gen.py", line 125, in simple_analysis
    caller = Caller(**caller_kwargs)
  File "/home/quinlan/.local/lib/python3.6/site-packages/ru/basecall.py", line 65, in __init__
    self.connect()
  File "/home/quinlan/.local/lib/python3.6/site-packages/pyguppyclient/client.py", line 73, in connect
    config = self._load_config(self.config_name)
  File "/home/quinlan/.local/lib/python3.6/site-packages/pyguppyclient/client.py", line 95, in _load_config
    loaded_configs = {Config(c).name: Config(c) for c in self.get_configs()}
  File "/home/quinlan/.local/lib/python3.6/site-packages/pyguppyclient/client.py", line 91, in get_configs
    res = self.send(SimpleRequestType.GET_CONFIGS)
  File "/home/quinlan/.local/lib/python3.6/site-packages/pyguppyclient/client.py", line 55, in send
    return simple_response(self.recv())
  File "/home/quinlan/.local/lib/python3.6/site-packages/pyguppyclient/client.py", line 65, in recv
    raise Again()
zmq.error.Again: Resource temporarily unavailable
[guppy/warning] main: The basecall server has been terminated.
```

**Comment by alexomics** [comment]
Brent, 

Very sorry for the slow response! I think that this issue is due to a guppy server incompatibility. We require guppy >=3.4 

If you are able to test with a newer guppy version (not one that ships with MinKNOW) and let me know, that would really help!

Alex

**Comment by brentp** [comment]
Hi Alex, thanks for the reply. I did have a newer guppy installed, and I'm pretty sure that's what was running. I won't be able to get back to this until quarantine is over, but I'll check then and get back to you.
-B

**Comment by tchrisboles** [comment]
Alex and Matt, 
If one has a linux box running new Minion-nc, AND new guppy (version 3.5.x) in a different directory (say home/local/bin), one could run basecalling "remote" using the local/bin copy in server mode as long as the toml file is properly configured (port other than 5556). Is that correct? I think that Alex had a diagram for some thing close to this in an earlier reply to someone setting up ru on a gridion. I want to set this up for a minion, and was wondering if running basecalling server on the same machine (but different installation directory) would work.  Any advice on setting up basecalling server or toml file for this case?

**Comment by alexomics** [comment]
@tchrisboles I have split this into another issue here: https://github.com/LooseLab/ru/issues/39
